### PR TITLE
Make sure scripts with symbols in path name work

### PR DESCRIPTION
**Pull request by coderabhishek**
This PR is intended as a patch for https://github.com/lihaoyi/Ammonite/issues/418.
Before this PR any script having special chars in path name fails because while loadingClasses wrapperNames are encoded whereas while findClass is given non-encoded Name as argument. 
This PR fixes this issue.

### CODE REVIEW

File: i..name.sc [hunk 1]
```diff
@@ -0,0 +1 @@
+println("Done")
```
Comment (lihaoyi) [inline i..name.sc]: This file looks un-used; can we kill it?

File: integration/src/test/resources/ammonite/integration/basic/script..run.sc [hunk 2]
```diff
@@ -0,0 +1 @@
+println("Script Worked!!")
```
Comment (lihaoyi) [inline integration/src/test/resources/ammonite/integration/basic/script..run.sc]: If we're going to test encoding of symbols, we might as well go the whole way and name the file `script%#.@*+叉燒.sc` and make sure all that gets properly handled

**Comment by lihaoyi** [comment]
I have verified that this fixes the problem, but could you add a unit test that replicates the original problem? i.e. run a script that lives outside of the working directory of the shell used to start the script? Ammonite-Ops lets you pass in a working directory when it runs external processes via `%%`.

File: integration/src/test/scala/ammonite/integration/BasicTests.scala [hunk 3]
```diff
@@ -25,6 +25,27 @@ object BasicTests extends TestSuite{
...
+          cwd/'integration/'src/'test/'resources/'ammonite/'integration/'basic/"script%#.@*+叉燒.sc"
+        rm(scriptAddr)
+        write(scriptAddr, """println("Script Worked!!")""")
+        val evaled = exec('basic / "script%#.@*+叉燒.sc")
+        assert(evaled.out.trim == "Script Worked!!" && evaled.err.string.isEmpty)
+      }
+    }
+
+    'scriptInSomeOtherDir{
+      rm(home/'bin/"script.sc")
```
Comment (lihaoyi) [inline integration/src/test/scala/ammonite/integration/BasicTests.scala]: Can use use a a `tmp.dir` instead of people's `home`?

**Comment by lihaoyi** [comment]
Just one tweak; when you're done squash it and I'll merge it

File: integration/src/test/scala/ammonite/integration/BasicTests.scala [hunk 4]
```diff
@@ -25,6 +25,27 @@ object BasicTests extends TestSuite{
       assert(evaled.out.trim == "Hello World")
     }
 
+    //make sure scripts with symbols in path names work fine
+    'scriptWithSymbols {
+      if (!Util.windowsPlatform){
+        val scriptAddr =
+          cwd/'integration/'target/'test/'resources/'ammonite/'integration/'basic/"script%#.@*+叉燒.sc"
+        rm(scriptAddr)
```
Comment (coderabhishek) [inline integration/src/test/scala/ammonite/integration/BasicTests.scala]: I have changed scriptAddr from src to target
I think this one should be better.

Comment (lihaoyi) [inline integration/src/test/scala/ammonite/integration/BasicTests.scala]: Sounds good

**Pull request merged**